framework/src/application/node/synchronizer/block_synchronization_mechanism.js

```diff
@@ -175,10 +174,8 @@ class BlockSynchronizationMechanism extends BaseSynchronizer {
 	async _handleBlockProcessingError(lastCommonBlock, peerId) {
 		// If the list of blocks has not been fully applied
 		this.logger.debug('Failed to apply obtained blocks from peer');
-		const [tipBeforeApplying] = await this.storage.entities.TempBlock.get(
-			{},
-			{ sort: 'height:desc', limit: 1, extended: true },
-		);
+		const tempBlocks = await this.blocks.dataAccess.getTempBlocks();
+		const [tipBeforeApplying] = tempBlocks.sort((a, b) => b - a);
```

Comment:
maybe not to mutate tempBlocks.. we could do `[...tempBlocks].sort((a, b) => b - a);`?